src/main/kotlin/no/nav/syfo/personstatus/infrastructure/database/PersonOversiktStatusRepository.kt

```diff
@@ -0,0 +1,128 @@
+import no.nav.syfo.personstatus.domain.PPersonOversiktStatus
+import no.nav.syfo.personstatus.domain.PersonOversiktStatus
+import no.nav.syfo.personstatus.domain.toPersonOversiktStatus
+import java.sql.ResultSet
+import java.sql.Timestamp
+import java.time.Instant
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.*
+
+class PersonOversiktStatusRepository(private val database: DatabaseInterface) : IPersonOversiktStatusRepository {
+
+    override fun updateArbeidsuforhetVurderingStatus(
+        personIdent: PersonIdent,
+        isAktivVurdering: Boolean
+    ): Result<Int> {
+        return try {
+            database.connection.use { connection ->
+                val tidspunkt = Timestamp.from(Instant.now())
+                val rowsUpdated = connection.prepareStatement(UPDATE_OR_INSERT_PERSON_OVERSIKT_STATUS).use {
+                    it.setString(1, UUID.randomUUID().toString())
+                    it.setString(2, personIdent.value)
+                    it.setBoolean(3, isAktivVurdering)
+                    it.setTimestamp(4, tidspunkt)
+                    it.setTimestamp(5, tidspunkt)
+                    it.executeUpdate()
+                }
+                val isSuccess = rowsUpdated == 1
+                if (isSuccess) {
+                    connection.commit()
+                    Result.success(rowsUpdated)
+                } else {
+                    connection.rollback()
+                    Result.failure(RuntimeException("Failed to update arbeidsuforhet vurdering status for person with fnr: ${personIdent.value}"))
+                }
+            }
+        } catch (e: Exception) {
+            Result.failure(e)
+        }
+    }
+
+    override fun getPersonOversiktStatus(personIdent: PersonIdent): PersonOversiktStatus? {
+        database.connection.use { connection ->
+            val personoversiktStatus = connection.prepareStatement(GET_PERSON_OVERSIKT_STATUS).use {
+                it.setString(1, personIdent.value)
+                it.executeQuery().toList { toPPersonOversiktStatus() }
+            }
+            return personoversiktStatus.firstOrNull()?.toPersonOversiktStatus()
+        }
+    }
+
+    companion object {
+        private const val GET_PERSON_OVERSIKT_STATUS =
+            """
+            SELECT *
+            FROM PERSON_OVERSIKT_STATUS
+            WHERE fnr = ?
+            """
+
+        private const val UPDATE_OR_INSERT_PERSON_OVERSIKT_STATUS =
+            """
+            INSERT INTO person_oversikt_status (
+                id,
+                uuid,
+                fnr,
+                arbeidsuforhet_aktiv_vurdering,
+                opprettet,
+                sist_endret
+            ) VALUES (DEFAULT, ?, ?, ?, ?, ?)
+            ON CONFLICT (fnr)
+            DO UPDATE SET
+                arbeidsuforhet_aktiv_vurdering = EXCLUDED.arbeidsuforhet_aktiv_vurdering,
+                sist_endret = EXCLUDED.sist_endret
```

Comment:
Så dette betyr: Hvis konflikt på fnr, oppdater disse kun disse to feltene med verdien vi sender inn?

Reply:
Stemmer! Dette `EXCLUDED` keywordet sånn jeg har skjønt det betyr at man bruker de nye verdiene, og ikke de som eksisterer i tabellen fra før 😊